automated_packaging/common_functions.pm

```diff
@@ -0,0 +1,146 @@
+        my @prs = @{ decode_json($prs_text) };
+
+        foreach my $pr (@prs) {
+            my %pr_hash = %$pr;
+
+            if ( defined( $pr_hash{'merged_at'} ) ) {
+
+                $merged_date = substr( $pr_hash{'merged_at'}, 0, 10 );
+
+                if ( $merged_date lt $earliest_pr_date ) {
+                    last;
+                }
+
+                $sorted_pr_hash{ $pr_hash{'merged_at'} } = $pr_hash{'url'};
+            }
+        }
+
+        $page_number += 1;
+    } while ( $merged_date gt $earliest_pr_date );
+
+    my @keys = reverse sort keys(%sorted_pr_hash);
+    my @vals = @sorted_pr_hash{@keys};
+
+    print( "PRs has been read in the merge order ..." . "\n" );
+
+    return @vals;
+}
+
+# Search for the backport label given a label array
+sub has_backport_label {
+
+    my @labels = @{ $_[0] };
+
+    foreach my $label (@labels) {
+        %label_hash = %{$label};
+        if ( $label_hash{'name'} eq 'backport' ) {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+# Creates changelog entries up to the given last date. It extracts lines starting with DESCRIPTION: from
+# PR messages to generate these entries. You can use different repo names to generate changelog entries for
+# different repos.
+sub create_release_changelog {
+    my $earliest_pr_date = @_[0];
+    my $repo_name = @_[1];
+    my $is_point_release = @_[2];
+
+    my @comment_lines = ();
+    my $github_token = get_and_verify_token();
+
+    my @sorted_pr_urls = get_sorted_prs( $earliest_pr_date, $repo_name );
+
+    foreach my $pr_url (@sorted_pr_urls) {
+        print( "Getting information for " . $pr_url . "\n" );
+        my $pr_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET --user "$github_token:x-oauth-basic" '$pr_url'`;
+        my %pr_hash = %{ decode_json($pr_text) };
+        my $add_to_changelog = 1;
+
+        if ($is_point_release) {
+            my $issue_url = $pr_hash{'issue_url'};
+            my $issue_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET --user "$github_token:x-oauth-basic" '$issue_url'`;
+            my %issue_hash = %{ decode_json($issue_text) };
+            $add_to_changelog = 0;
+
+            @labels = @{ $issue_hash{'labels'} };
+            if ( has_backport_label( \@labels ) ) {
+                $add_to_changelog = 1;
+            }
+        }
+
+        if ($add_to_changelog) {
+            @log_output = split( "\n", $pr_hash{'body'} );
+            foreach $line (@log_output) {
+                if ( $line =~ /^DESCRIPTION:.*/ ) {
+                    $description_part = substr( $line, 13, -1 );
+                    if ( length($description_part) > 78 ) {
+                        print(  "You have to shorten PR message '"
+                              . $description_part . "' of "
```

Comment:
I believe we can directly use variables in double quotes. No need to end quote and concatenate with dots. After that one line would be enough for whole message I guess.